generators/java/index.cjs

```diff
@@ -0,0 +1,212 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable consistent-return */
+const chalk = require('chalk');
+const { mixBlueprintGenerator } = require('generator-jhipster/support');
+
+const { GENERATOR_PROJECT_NAME, GENERATOR_INIT, GENERATOR_JAVA } = require('../generator-list');
+const { PACKAGE_NAME, BUILD_TOOL, PRETTIER_JAVA_INDENT } = require('./constants.cjs');
+const { files } = require('./files.cjs');
+const { defaultConfig } = require('./config.cjs');
+
+const MixedGenerator = mixBlueprintGenerator(GENERATOR_PROJECT_NAME, GENERATOR_INIT, GENERATOR_JAVA);
+
+module.exports = class extends MixedGenerator {
+  constructor(args, opts, features) {
+    super(args, opts, { jhipsterModular: true, unique: 'namespace', ...features });
+
+    // Register options available to cli.
+    if (!this.fromBlueprint) {
+      this.registerCommonOptions();
+      this.registerProjectNameOptions();
+      this.registerInitOptions();
+      this.registerJavaOptions();
+    }
+
+    if (this.options.help) return;
+
+    if (this.options.defaults) {
+      this.configureProjectName();
+      this.configureInit();
+      this.configureJava();
+    }
+  }
+
+  async _beforeQueue() {
+    if (!this.fromBlueprint) {
+      await this.dependsOnJHipster(GENERATOR_PROJECT_NAME);
+      if (this.shouldComposeModular()) {
+        await this.dependsOnJHipster(GENERATOR_INIT);
+      }
+      await this.composeWithBlueprints(GENERATOR_JAVA);
+    }
+  }
+
+  _initializing() {
+    return {
+      validateFromCli() {
+        this.checkInvocationFromCLI();
+      },
+      sayHello() {
+        if (!this.showHello()) return;
+        this.log(chalk.white('⬢ Welcome to the JHipster Java ⬢'));
+      },
+      loadRuntimeOptions() {
+        this.loadRuntimeOptions();
+      },
+      loadOptionsConstants() {
+        this.loadJavaOptionsConstants();
+      },
+    };
+  }
+
+  get initializing() {
+    if (this.delegateToBlueprint) return;
+    return this._initializing();
+  }
+
+  _prompting() {
+    return {
+      async showPrompts() {
+        if (this.shouldSkipPrompts()) return;
+        await this.prompt(
+          [
+            {
+              name: PACKAGE_NAME,
+              type: 'input',
+              validate: input => this._validatePackageName(input),
+              message: 'What is your default Java package name?',
+              default: () => this._getDefaultPackageName(),
+            },
+            {
+              name: BUILD_TOOL,
```

Comment:
prompts.js is possible but I don't think it's too useful.
It would require to pass the generator as parameter or bind context to the function.

Reply:
Just to avoid the previous generator in the chain to reference properties/prompts of the next ones. So that contributor will know were to find the information, and generators will be really independent together (not bound by a parent->child relationship)